### PR TITLE
Add deletion_protection field to Metastore Service

### PR DESCRIPTION
**Pull request by SamanthaMathews**
Add deletion_protection field to make deletion actions require an explicit intent
Part of b/366197455

Part of https://github.com/hashicorp/terraform-provider-google/issues/18854

**Release Note Template for Downstream PRs (will be copied)**
```
Metastore: added `deletion_protection` field to `dataproc_metastore_service` to make deleting them require an explicit intent. `google_active_directory_domain` resources now cannot be destroyed unless `deletion_protection = false` is set for the resource.
```

### CODE REVIEW

**Comment by c2thorn** [comment]
Discussing with @SamanthaMathews offline about this PR and the go rewrite migration

File: mmv1/third_party/terraform/services/dataprocmetastore/go/resource_dataproc_metastore_service_test.go [hunk 1]
```diff
@@ -33,17 +35,49 @@ func TestAccDataprocMetastoreService_updateAndImport(t *testing.T) {
...
+                                ImportState:             true,
+                                ImportStateVerify:       true,
+                                ImportStateVerifyIgnore: []string{"deletion_protection"},
+                        },
+                        {
+                                Config: testAccDataprocMetastoreService_updateLocation_deletionProtection(name, "us-west2", tier[1]),
+                                ExpectError: regexp.MustCompile("deletion_protection"),
+                        },
+                        {
+                                Config: testAccDataprocMetastoreService_updateAndImport(name, tier[1]),
```
Comment (abd-goog) [inline mmv1/third_party/terraform/services/dataprocmetastore/go/resource_dataproc_metastore_service_test.go:68]: Should be tier[0] I believe to match step 1 resource.

**Review by abd-goog — CHANGES_REQUESTED** [review]

File: mmv1/products/metastore/Service.yaml [hunk 2]
```diff
@@ -53,6 +53,16 @@ async: !ruby/object:Api::OpAsync
     path: 'error'
     message: 'message'
 autogen_async: true
+virtual_fields:
+  - !ruby/object:Api::Type::Boolean
+    name: 'deletion_protection'
+    default_value: true
+    description: |
+      Whether Terraform will be prevented from destroying the service. Defaults to false.
```
[the remark this answers is not in the document]
Comment (SamanthaMathews) [reply mmv1/products/metastore/Service.yaml:61]: default value is false

File: mmv1/products/metastore/Service.yaml [hunk 3]
```diff
@@ -53,6 +53,16 @@ async: !ruby/object:Api::OpAsync
...
+      When the field is set to true in Terraform state, a `terraform apply`
+      or `terraform destroy` that would delete the service will fail.
```
[the remark this answers is not in the document]
Comment (SamanthaMathews) [reply mmv1/products/metastore/Service.yaml:63]: default value is false

**Review by abd-goog — APPROVED** [review]

**Comment by c2thorn** [comment]
Hi Samantha, please recreate this PR per our conversation earlier this month.

**Pull request closed without merging**